os/arch/arm/src/s5j/s5j_idle.c

```diff
@@ -161,6 +165,11 @@ void up_idle(void)
 	/* Perform IDLE mode power management */
 	up_idlepm();
 
+#if !defined CONFIG_PM && defined CONFIG_SCHED_TICKSUPPRESS
```

Comment:
I don't understand, why CONFIG_PM flag has been checked  here ? whats the dependency of this feature with CONFIG_PM ?  I believe these are independent modules . isn't it ?

Reply:
Existing PM framework is based on periodic timer tick interrupts. If we suppress timer tick interrupts, then PM algorithm doesn't work.
So I have handled that case as follows,
1) If ```CONFIG_PM``` is not enabled  and ```CONFIG_SCHED_TICKSUPPRESS```  is enabled
---> apply tick suppression immediately.

2) If ```CONFIG_PM``` is enabled and ```CONFIG_SCHED_TICKSUPPRESS```  is also enabled
---> apply tick suppression at the last state of PM states i.e. after PM algorithm completes.